core_lib/src/canvaspainter.cpp

```diff
@@ -543,6 +543,69 @@ void CanvasPainter::paintGrid(QPainter& painter)
+    for (int i = 0; i < mObject->getLayerCount(); ++i)
+    {
+        Layer* layer = mObject->getLayer(i);
+        if (layer->type() == Layer::CAMERA && layer->visible())
+        {
+            isCameraMode = (i == mCurrentLayerIndex);
+            break;
+        }
+    }
+
+    QPen pen(Qt::black);
+    pen.setCosmetic(true);
+    painter.setPen(pen);
+    painter.setWorldMatrixEnabled(!isCameraMode);
+    painter.setBrush(Qt::NoBrush);
+    QPainter::RenderHints previous_renderhints = painter.renderHints();
+    painter.setRenderHint(QPainter::Antialiasing, false);
+
+    switch (ol) {
+    case CENTER:
+        painter.drawLine(rect.x() + rect.width()/4, rect.y() + rect.height()/2, rect.x() + rect.width()*3/4, rect.y() + rect.height()/2);
+        painter.drawLine(rect.x() + rect.width()/2, rect.y() + rect.height()/4, rect.x() + rect.width()/2, rect.y() + rect.height()*3/4);
+        break;
+    case THIRDS:
+        painter.drawLine(rect.x(), rect.y() + (rect.height()/3), rect.right(), rect.y() + (rect.height()/3));
+        painter.drawLine(rect.x(), rect.y() + (rect.height() * 2/3), rect.x() + rect.width(), rect.y() + (rect.height() * 2/3));
+        painter.drawLine(rect.x() + rect.width()/3, rect.y(), rect.x() + rect.width()/3, rect.y() + rect.height());
+        painter.drawLine(rect.x() + rect.width() *2/3, rect.y(), rect.x() + rect.width() *2/3, rect.y() + rect.height());
+        break;
+    case GOLDEN:
+        painter.drawLine(rect.x(), static_cast<int>(rect.y() + (rect.height() * 0.38)), rect.right(), static_cast<int>(rect.y() + (rect.height() * 0.38)));
+        painter.drawLine(rect.x(), static_cast<int>(rect.y() + (rect.height() * 0.62)), rect.x() + rect.width(), static_cast<int>(rect.y() + (rect.height() * 0.62)));
+        painter.drawLine(static_cast<int>(rect.x() + rect.width() * 0.38), rect.y(), static_cast<int>(rect.x() + rect.width() * 0.38), rect.bottom());
+        painter.drawLine(static_cast<int>(rect.x() + rect.width() * 0.62), rect.y(), static_cast<int>(rect.x() + rect.width() * 0.62), rect.bottom());
+        break;
+    case SAFE:
+        if (mOptions.bActionSafe)
+        {
+            int action = mOptions.nActionSafe;
+            QRect safeAction = QRect(rect.x() + rect.width()*action/200, rect.y() + rect.height()*action/200, rect.width()*(100-action)/100, rect.height()*(100-action)/100);
+            painter.drawRect(safeAction);
+            painter.drawText(safeAction.x(), safeAction.y(), tr("Safe Action area %1 %").arg(action));
+        }
+        if (mOptions.bTitleSafe)
+        {
+            int title = mOptions.nTitleSafe;
+            QRect safeTitle = QRect(rect.x() + rect.width()*title/200, rect.y() + rect.height()*title/200, rect.width()*(100-title)/100, rect.height()*(100-title)/100);
+            painter.drawRect(safeTitle);
+            painter.drawText(safeTitle.x(), safeTitle.y(), tr("Safe Title area %1 %").arg(title));
+        }
+        break;
+    }
```

Comment:
This could be split up via private methods.

Reply:
Yes, it could, but the four overlays share 20 lines of code that must be repeated in every function, if they are split up.
On the other hand, I've thought of using if(...) in stead of a switch, and call from renderOverlays, if minimum one overlay is active. In this case, it would only be with painter as parameter.

Reply:
I have optimized the code, but kept it in one function. Should I split it in four?

Reply:
Split up in four functions. Done.